metrics/src/control.rs

```diff
@@ -1,53 +1,49 @@
-use super::data::snapshot::Snapshot;
-use crossbeam_channel::{bounded, Sender};
+use crate::data::Snapshot;
+use crate::registry::{MetricRegistry, ScopeRegistry};
 use futures::prelude::*;
 use metrics_core::{AsyncSnapshotProvider, SnapshotProvider};
 use std::error::Error;
 use std::fmt;
-use tokio_sync::oneshot;
+use std::sync::Arc;
 
-/// Error conditions when retrieving a snapshot.
+/// Error during snapshot retrieval.
 #[derive(Debug, Clone)]
 pub enum SnapshotError {
```

Comment:
same non exahustive stuff as above to support forward compat

Reply:
Done.